.github/workflows/build.yml

```diff
@@ -0,0 +1,180 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  DOCKER_HOSTNAME: ghcr.io
+  DOCKER_USERNAME: the-mesh-for-data
+  ACTIVE_CI: travisci # This parameter defines which CI should push to the docker repository
+
+jobs:
+  verify:
+    name: Verify
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_HOSTNAME: "localhost:5000"
+      DOCKER_NAMESPACE: m4d-system
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.13
+    - uses: actions/checkout@v2
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.m2/repository
+          ~/go/pkg/mod
+          ${{ github.workspace }}/hack/tools/bin
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: ${{ runner.os }}-go
+    - name: Verify
+      run: make verify
```

Comment:
I agree. Do you think git hooks would be an option for e.g. `go fmt` ?

Reply:
A commit hook? You would need to trust people to set it up after cloning the repo. For CI we can check if go fmt did anything and fail otherwise (like what https://github.com/grandcolline/golang-github-actions does). As a developer I just set my IDE to do.